tests/suites/os/tests/config-json/index.js

```diff
@@ -236,37 +236,48 @@ module.exports = {
 					uri: 'http://www.archlinux.org/check_network_status.txt',
 				};
 
-				await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						`tmp=$(mktemp)&&cat /mnt/boot/config.json | jq '.os.network.connectivity=${JSON.stringify(
-							connectivity,
-						)}' > $tmp&&mv "$tmp" /mnt/boot/config.json`,
-						this.context.get().link,
-					);
-
-				// Rebooting the DUT
-				await this.context.get().worker.rebootDut(this.context.get().link);
+				const context = this.context.get();
 
-				const config = await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						'NetworkManager --print-config | awk "/\\[connectivity\\]/{flag=1;next}/\\[/{flag=0}flag"',
-						this.context.get().link,
+				test.comment('Configuring connectivity check in config.json')
+				return context.worker.executeCommandInHostOS(
+						[
+							'tmp=$(mktemp)',
+							'&&', 'jq', `'.os.network.connectivity=${JSON.stringify(connectivity)}'`,
+								'/mnt/boot/config.json',
+							'>', '$tmp', '&&', 'mv', '$tmp', '/mnt/boot/config.json',
+						].join(' '),
+						context.link,
+				).then(() => {
+					test.comment('Restarting os-networkmanager.service');
+					return context.worker.executeCommandInHostOS(
+						'systemctl restart os-networkmanager.service',
```

Comment:
So as a general rule we are saying the following:

> Restarting services manually in lieu of a manual reboot is safe for testing and close enough to real use cases

I think I'm okay with this, as long as it's not something that could be impacted by a race condition during reboots?

Reply:
@klutchell It's a fair point, they _are_ testing different things. I think whether or not this is acceptable depends on the scope of what we're trying to test. In this particular instance, we know that the service requires `bind-etc-NetworkManager-conf.d.service` in order to write config fragments to the proper location. I think it's safe to assume systemd is well-tested, so it makes little sense to test systemd functionality. We could probably explicitly check for that service dependency (using, e.g., `systemctl show ...`) instead of rebooting and hoping we see a race condition manifest as a flaky test. This way, if the dependency is ever removed, renamed, or modified, we get a test failure to warn us.

Reply:
I think I would prefer we just fail the functional test if a dependency doesn't run or is no longer defined. It seems needlessly complicated to inspect each service to see that they all have the dependencies we expect.

I think what you have is good, and maybe even better than a reboot because we may find restarting the service manually doesn't work but a reboot does. That's a good indication that we haven't added a dependency somewhere.